src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/Scripts.cs

```diff
@@ -41,5 +43,29 @@ public Scripts(Compiler compiler)
             }
             return null;
         }
+
+        internal class TrimSafeDictionary
+        {
+            private readonly Dictionary<string, Type?> _backingDictionary = new Dictionary<string, Type?>();
+
+            public Type? this[string key]
+            {
+                [UnconditionalSuppressMessage("TrimAnalysis", "IL2073:MissingDynamicallyAccessedMembers",
+                    Justification = "The getter of the dictionary is not annotated to preserve the constructor, but the sources that are adding the items to " +
+                    "the dictionary are annotated so we can supress the message as we know the constructor will be preserved.")]
+                [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+                get => ((IDictionary<string, Type?>)_backingDictionary)[key];
+                [param: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
+                set => ((IDictionary<string, Type?>)_backingDictionary)[key] = value;
+            }
+
+            public ICollection<string> Keys => ((IDictionary<string, Type?>)_backingDictionary).Keys;
+
+            public int Count => ((ICollection<KeyValuePair<string, Type?>>)_backingDictionary).Count;
```

Comment:
Why are all these casts of `_backingDictionary` necessary? Can't you just return `_backingDictionary.Count`?

Reply:
I'll fix it, not necessary, I just used the  quick refactoring ctrl+'.' options in VS that added the casts:
![image](https://user-images.githubusercontent.com/13854455/111702612-c8c6a480-87f9-11eb-8167-763daa2bfbaa.png)